apps/lockfile-explorer-web/src/containers/PackageJsonViewer/index.tsx

```diff
@@ -9,47 +15,133 @@ enum PackageView {
 }
 
 export const PackageJsonViewer = (): JSX.Element => {
-  const [packageJSON, setPackageJSON] = useState('');
-  const [parsedPackageJSON, setParsedPackageJSON] = useState('');
+  const [packageJSON, setPackageJSON] = useState<IPackageJson | undefined>(undefined);
+  const [parsedPackageJSON, setParsedPackageJSON] = useState<IPackageJson | undefined>(undefined);
+  const [specChanges, setSpecChanges] = useState<Map<string, ISpecChange>>(new Map());
   const [cjs, setCjs] = useState('');
+  const selectedEntry = useAppSelector(selectCurrentEntry);
 
-  const [selection, setSelection] = useState<PackageView>(PackageView.PACKAGE_JSON);
+  const [selection, setSelection] = useState<PackageView>(PackageView.PARSED_PACKAGE_JSON);
 
   const cb = useCallback((s: PackageView) => () => setSelection(s), []);
 
   useEffect(() => {
-    async function loadPackageDetails(): Promise<void> {
-      const cjsFile = await readCJS();
+    async function loadPackageDetails(packageName: string): Promise<void> {
+      const cjsFile = await readPnpmfile();
       setCjs(cjsFile);
-      const packageJSONFile = await readPackageJSON();
-      setPackageJSON(packageJSONFile as string);
-      const parsedJSON = await readParsedCJS();
-      setParsedPackageJSON(parsedJSON as string);
+      const packageJSONFile = await readPackageJSON(packageName);
+      setPackageJSON(packageJSONFile);
+      const parsedJSON = await readParsedCJS(packageName);
+      setParsedPackageJSON(parsedJSON);
+
+      if (packageJSONFile && parsedJSON) {
+        const diffDeps = compareSpec(packageJSONFile, parsedJSON);
+        setSpecChanges(diffDeps);
+      }
+    }
+    if (selectedEntry) {
+      if (selectedEntry.entryPackageName) {
+        /* eslint @typescript-eslint/no-floating-promises: off */
+        loadPackageDetails(selectedEntry.packageJsonFolderPath);
```

Comment:
The ESLint rule `@typescript-eslint/no-floating-promises` generally should not be suppressed unless you can prove that the function cannot possibly reject its promise.  

This situation commonly occurs when a synchronous function is starting an asynchronous operation. Generally the fix is to add a `.catch((e) => { _______ });` handler, ideally reporting the error somehow.  (For example, if the user clicks a `Submit` button and the service returned 403, an app shouldn't just sit there unresponsively without telling the user that something went wrong.)

Reply:
What if I handle the error in the asynchronous function? For example adding a try catch block to handle the error inside the loadPackageDetails function - would that accomplish the same thing?

Reply:
I don't think the ESLint rule will be smart enough to realize that the `try`-`catch` covers every possible statement in the code block.  I suspect they want `.catch()` because it's explicit and easy to check.   See here: https://typescript-eslint.io/rules/no-floating-promises/

Reply:
changed!